### PR TITLE
Implement CRUD Operation Endpoints for User, Customer, Consumer, RerservationTransaction, Reservation, Payment Entities

### LINKED ISSUE #3: Implement CRUD Operation Endpoints for User, Customer, Consumer, RerservationTransaction, Reservation,  Payment Entities
This issue involves creating CRUD functionality for the following entities within the web application.
- [x] User
- [x] Customer 
- [x] Consumer
- [x] RerservationTransaction
- [x] Reservation
- [x] Payment

**Actions**
- Create
- Edit
- Delete
- Update

### PR DESCRIPTION
**Pull request by spatiszisis**
- Removed the flyway implementation. We will introduce it in a future issue.
- Introduced a basic CRUD system for the entities (User, Customer, Consumer, ReservationTransaction, Reservation, and Payment).
- Added an exception handler.

### CODE REVIEW

**Review by apokiou — APPROVED** [review]
Looks Good !

**Review by EmmanSkout — APPROVED** [review]
Very good , a minor (maybe) point is that we can possible have a base abstract controller in the future

File: src/main/java/net/trapezokomos/dashboard/service/IAbstractService.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package net.trapezokomos.dashboard.service;
+
+import net.trapezokomos.dashboard.exception.ResourceAlreadyExistsException;
+import org.springframework.data.domain.Pageable;
+
+public interface IAbstractService<T, ID> {
```
Comment (christosavramis) [inline src/main/java/net/trapezokomos/dashboard/service/IAbstractService.java]: remove T, replace ID with Entity

File: src/main/java/net/trapezokomos/dashboard/data/BaseDocument.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import jakarta.persistence.*;
+import lombok.Data;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.LastModifiedDate;
+
+import java.util.Date;
+
+@Data
+@MappedSuperclass
+public class BaseDocument {
```
Comment (christosavramis) [inline src/main/java/net/trapezokomos/dashboard/data/BaseDocument.java]: Rename to AbstractEntity

File: src/main/java/net/trapezokomos/dashboard/data/User.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+@Table(
+        name = "users",
+        uniqueConstraints = {
+                @UniqueConstraint(name = "user_email_unique", columnNames = "email"),
+                @UniqueConstraint(name = "user_phone_unique", columnNames = "phoneNumber"),
+        }
+)
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
```
Comment (christosavramis) [inline src/main/java/net/trapezokomos/dashboard/data/User.java]: @Data may cause performance issues,
use
@Setter @GETTER

File: src/main/java/net/trapezokomos/dashboard/exception/ResourceAlreadyExistsException.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package net.trapezokomos.dashboard.exception;
+
+import java.io.Serial;
+
+public class ResourceAlreadyExistsException extends Exception {
```
Comment (christosavramis) [inline src/main/java/net/trapezokomos/dashboard/exception/ResourceAlreadyExistsException.java]: Lets create an intermediate Exception Class 

GenericAPIException
GenericAPIRunTimeException
and extend through those

File: db_schema.puml [hunk 5]
```diff
@@ -0,0 +1,133 @@
+@startuml
```
Comment (christosavramis) [inline db_schema.puml:1]: why we need this?

File: src/main/java/net/trapezokomos/dashboard/utils/UserConverter.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import net.trapezokomos.dashboard.data.Role;
+import net.trapezokomos.dashboard.data.User;
+import net.trapezokomos.dashboard.resources.UserResource;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+import java.util.Set;
+
+@Component
+public class UserConverter implements AbstractConverter<UserResource, User> {
```
Comment (christosavramis) [inline src/main/java/net/trapezokomos/dashboard/utils/UserConverter.java]: We could also try spring converters interface Converter

File: src/main/java/net/trapezokomos/dashboard/controller/UserController.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/user")
```
Comment (christosavramis) [inline src/main/java/net/trapezokomos/dashboard/controller/UserController.java]: lets move the api to "api/v1/user" to be able to easily serve html on root or not and have api versioning

File: db_schema.puml (hunk 5 above)
Comment (spatiszisis) [reply db_schema.puml:1]: This is a structure annotation for **.puml** files to know the plugin that reads these files that this is the start of the specific file

**Pull request merged**